pyfar/dsp/dsp.py

```diff
@@ -561,6 +563,116 @@ def time_window(signal, interval, window='hann', shape='symmetric',
         return signal_win
 
 
+def time_crop(signal, interval: Union[list[float], tuple[float, float],
+                                       np.ndarray],
+              unit: Literal["samples", "s"]='samples'):
+    """Crop a pf.Signal or pf.TimeData object in time.
+
+    Parameters
+    ----------
+    signal : pyfar.Signal, pyfar.TimeData
+        pf.Signal or pf.TimeData object to be cropped.
+
+    interval : list, tuple, or numpy.ndarray
+        Must contain exactly two entries, these specify the beginning and
+        the end of the section to be cropped. The unit of `interval`
+        is specified by the parameter `unit`.
+
+    unit : string, optional
+        Unit of `interval`. Can be set to ``'samples'`` or ``'s'`` (seconds).
+        Time values are rounded to the nearest sample. The default is
+        ``'samples'``.
+
+    Returns
+    -------
+    cropped_signal : pyfar.Signal, pyfar.TimeData
+        Cropped pf.Signal or pf.TimeData object.
+
+    Examples
```

Comment:
Very nice example! I would suggest to split it up into two, one for a Signal and one for TimeData (each with two panels in the plot). A short sentence before each example could highlight the difference.